prusti/src/callbacks.rs

```diff
@@ -21,14 +21,19 @@ pub struct PrustiCompilerCalls;
 
 #[allow(clippy::needless_lifetimes)]
 fn mir_borrowck<'tcx>(tcx: TyCtxt<'tcx>, def_id: LocalDefId) -> mir_borrowck<'tcx> {
-    let body_with_facts = prusti_rustc_interface::borrowck::consumers::get_body_with_borrowck_facts(
-        tcx,
-        ty::WithOptConstParam::unknown(def_id),
-    );
-    // SAFETY: This is safe because we are feeding in the same `tcx` that is
-    // going to be used as a witness when pulling out the data.
-    unsafe {
-        mir_storage::store_mir_body(tcx, def_id, body_with_facts);
+    // Running `get_body_with_borrowck_facts` can be very slow, therefore we avoid it when not necessary:
+    // for crates which won't be verified or spec_fns it suffices to load just the fn body
+    if !config::no_verify() && !is_spec_fn(tcx, def_id.to_def_id()) {
```

Comment:
Have you considered moving the `if !config::no_verify()` condition around the `config.override_queries = Some(override_queries);` statement?